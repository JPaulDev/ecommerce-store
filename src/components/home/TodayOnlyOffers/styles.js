import styled, { css } from 'styled-components';

const flexStyles = css`
  align-items: center;
  display: flex;
`;

export const Container = styled.div`
  background-color: white;
  border: 1px solid var(--color-grey-9);
  margin-bottom: 15px;
  padding: 15px 5px;
`;

export const List = styled.ul`
  justify-content: space-around;
  list-style: none;
  margin: 0 auto;
  max-width: 1100px;
  ${flexStyles}
`;

export const Button = styled.button`
  background: none;
  cursor: pointer;
  flex-direction: column;
  min-width: 85px;
  ${flexStyles}

  @media screen and (min-width: 640px) {
    min-width: 95px;
  }
`;

export const Wrapper = styled.div`
  border-radius: 50%;
  box-shadow: 0 0 0 1px var(--color-orange-1);
  height: 65px;
  justify-content: center;
  margin-bottom: 5px;
  position: relative;
  width: 65px;
  ${flexStyles}

  &::after {
    border-radius: inherit;
    box-shadow: 0 0 0 2px var(--color-orange-1);
    content: '';
    height: 100%;
    opacity: 0;
    position: absolute;
    transition: opacity 200ms ease-in-out;
    width: 100%;

    ${Button}:hover & {
      opacity: 1;
    }
  }

  @media screen and (min-width: 550px) {
    width: 70px;
    height: 70px;
  }

  @media screen and (min-width: 640px) {
    width: 80px;
    height: 80px;
  }
`;

export const Heading = styled.h3`
  color: var(--color-grey-5);
  font-size: 0.75rem;
  font-weight: 600;
  transition: color 200ms ease-in-out;

  ${Button}:hover & {
    color: var(--color-grey-2);
  }

  @media screen and (min-width: 640px) {
    font-size: 0.84rem;
  }
`;

export const ProductList = styled.ul`
  display: grid;
  gap: 10px;

  @media screen and (min-width: 570px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media screen and (min-width: 840px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (min-width: 1110px) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (max-width: 1310px) {
    padding: 0 10px;
  }
`;
