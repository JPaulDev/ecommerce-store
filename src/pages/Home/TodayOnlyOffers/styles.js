import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  margin-top: 8px;

  @media screen and (max-width: 580px) {
    flex-direction: column;
    row-gap: 8px;
    padding: 0 20px;
  }

  @media screen and (max-width: 1310px) {
    padding: 0 10px;
  }
`;
