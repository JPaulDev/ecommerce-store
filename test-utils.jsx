import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { setupStore } from './src/app/store';
import { UIProvider } from './src/contexts/UIContext';
import theme from './src/styles/theme';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <UIProvider>{children}</UIProvider>
        </Provider>
      </ThemeProvider>
    );
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
export { renderWithProviders as render };
