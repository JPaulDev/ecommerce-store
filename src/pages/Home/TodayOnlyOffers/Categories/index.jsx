import { ReactComponent as GfxIcon } from '../../../../assets/svg/gfx.svg';
import { ReactComponent as MemoryIcon } from '../../../../assets/svg/memory.svg';
import { ReactComponent as HddIcon } from '../../../../assets/svg/hdd.svg';
import { ReactComponent as ProcessorIcon } from '../../../../assets/svg/processor.svg';
import { ReactComponent as MotherboardIcon } from '../../../../assets/svg/motherboard.svg';
import CategoryItem from './CategoryItem';
import * as Styled from './styles';

const categories = [
  {
    heading: 'Graphics Cards',
    icon: GfxIcon,
  },
  {
    heading: 'Memory',
    icon: MemoryIcon,
  },
  {
    heading: 'Hard Drives',
    icon: HddIcon,
  },
  {
    heading: 'Processors',
    icon: ProcessorIcon,
  },
  {
    heading: 'Motherboards',
    icon: MotherboardIcon,
  },
];

export default function Categories({ onChangeCategory }) {
  return (
    <Styled.Container>
      <Styled.List>
        {categories.map((category) => (
          <CategoryItem
            key={category.heading}
            Icon={category.icon}
            heading={category.heading}
            onChangeCategory={() => onChangeCategory(category.heading)}
          />
        ))}
      </Styled.List>
    </Styled.Container>
  );
}
