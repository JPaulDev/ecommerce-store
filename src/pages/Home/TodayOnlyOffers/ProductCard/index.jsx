import { useState, forwardRef } from 'react';
import usePrice from '../../../../hooks/usePrice';
import Price from '../../../../components/Price';
import PreviousPrice from '../../../../components/PreviousPrice';
import PartNumber from '../../../../components/PartNumber';
import QuantityDropdown from '../../../../components/QuantityDropdown';
import StockIndicator from '../../../../components/StockIndicator';
import AddToBasketButton from '../../../../components/AddToBasketButton';
import * as Styled from './styles';

const ProductCard = forwardRef(({ product }, ref) => {
  const [quantity, setQuantity] = useState(1);
  const { price, salePrice } = usePrice(product);

  const handleChangeQuantity = (e) => setQuantity(parseInt(e.target.value, 10));

  return (
    <Styled.Wrapper ref={ref}>
      <Styled.Container>
        <Styled.ProductImage
          src={product.image}
          alt=""
          width={200}
          height={200}
        />
        <Styled.ProductName>{product.name}</Styled.ProductName>
        <Styled.ProductDescription>
          {product.description.join(', ')}
        </Styled.ProductDescription>
        <Styled.PartNumberWrapper>
          <PartNumber $fontSize="0.75rem">{product.sku}</PartNumber>
        </Styled.PartNumberWrapper>
        <Price price={salePrice || price} fontSize="1.9rem" />
        {salePrice && <PreviousPrice price={price} fontSize="0.93rem" />}
        <QuantityDropdown
          quantity={quantity}
          isDisabled={!product.stock}
          onChangeQuantity={handleChangeQuantity}
        />
        <StockIndicator stock={product.stock} marginTop="8px" />
        <AddToBasketButton
          type="button"
          disabled={!product.stock}
          $width="160px"
          $fontSize="0.75rem"
        >
          Add To Basket
        </AddToBasketButton>
      </Styled.Container>
    </Styled.Wrapper>
  );
});

export default ProductCard;
