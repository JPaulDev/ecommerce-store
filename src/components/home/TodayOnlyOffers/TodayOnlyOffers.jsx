import { useState } from 'react';
import { AnimatePresence, motion } from 'framer-motion';
import {
  GraphicsCard,
  Memory,
  HardDrive,
  Processor,
  Motherboard,
} from '../../icons';
import ProductCard from './ProductCard';
import * as Styled from './styles';

const categories = [
  {
    heading: 'Graphics Cards',
    icon: <GraphicsCard width="60%" />,
    id: 4,
  },
  {
    heading: 'Memory',
    icon: <Memory width="60%" />,
    id: 2,
  },
  {
    heading: 'Storage',
    icon: <HardDrive width="60%" />,
    id: 7,
  },
  {
    heading: 'Processors',
    icon: <Processor width="60%" />,
    id: 1,
  },
  {
    heading: 'Motherboards',
    icon: <Motherboard width="60%" />,
    id: 3,
  },
];

const MotionProductCard = motion(ProductCard);
const motionProps = {
  variants: {
    visible: {
      opacity: 1,
    },
    hidden: {
      opacity: 0,
    },
  },
  transition: {
    ease: 'easeInOut',
    duration: 0.6,
  },
};

export default function TodayOnlyOffers({ products = [] }) {
  const [shopCategory, setShopCategory] = useState(1);

  const handleChangeCategory = (categoryId) => setShopCategory(categoryId);

  const filteredProducts = products.filter(
    (product) => product.parentCategoryId === shopCategory
  );

  return (
    <section>
      <Styled.Container>
        <Styled.List>
          {categories.map((category) => (
            <li key={category.id}>
              <Styled.Button
                type="button"
                onClick={() => handleChangeCategory(category.id)}
              >
                <Styled.Wrapper>{category.icon}</Styled.Wrapper>
                <Styled.Heading>{category.heading}</Styled.Heading>
              </Styled.Button>
            </li>
          ))}
        </Styled.List>
      </Styled.Container>
      <Styled.ProductList>
        <AnimatePresence initial={false} mode="wait">
          {filteredProducts.map((product) => (
            <MotionProductCard
              key={product.sku}
              product={product}
              initial="hidden"
              animate="visible"
              exit="hidden"
              {...motionProps}
            />
          ))}
        </AnimatePresence>
      </Styled.ProductList>
    </section>
  );
}
