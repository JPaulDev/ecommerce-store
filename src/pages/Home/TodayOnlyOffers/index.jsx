import { useState, useEffect, useRef } from 'react';
import { AnimatePresence, motion } from 'framer-motion';
import useMediaQuery from '../../../hooks/useMediaQuery';
import offers from '../../../data/today-only-offers';
import Categories from './Categories';
import ProductCard from './ProductCard';
import * as Styled from './styles';

const MotionProductCard = motion(ProductCard);

function setInitialProducts() {
  let productsPerRow = 2;

  if (window.innerWidth >= 1120) {
    productsPerRow = 4;
  } else if (window.innerWidth >= 845) {
    productsPerRow = 3;
  }

  return productsPerRow;
}

// When changing category disables all other transitions except for opacity
const duration = 0.5;
const variants = {
  enter: (isChangingCategory) => ({
    opacity: 0,
    flex: isChangingCategory ? 1 : 0,
    margin: isChangingCategory ? '0 4px' : 0,
  }),
  visible: (isChangingCategory) => ({
    opacity: 1,
    flex: 1,
    margin: '0 4px',
    transition: {
      ease: 'easeInOut',
      duration,
      opacity: {
        duration,
        delay: isChangingCategory ? 0 : duration,
      },
    },
  }),
  exit: (isChangingCategory) => ({
    opacity: 0,
    flex: isChangingCategory ? 1 : 0,
    margin: isChangingCategory ? '0 4px' : 0,
    transition: {
      ease: 'easeInOut',
      duration,
      flex: {
        duration,
        delay: isChangingCategory ? 0 : duration,
      },
      margin: {
        duration,
        delay: isChangingCategory ? 0 : duration,
      },
    },
  }),
};

export default function TodayOnlyOffers() {
  const [displayCategory, setDisplayCategory] = useState('processors');
  const [productsPerRow, setProductsPerRow] = useState(setInitialProducts);
  const isMatch845 = useMediaQuery('(min-width: 845px)');
  const isMatch1120 = useMediaQuery('(min-width: 1120px)');
  const isChangingCategory = useRef(false);

  const handleChangeCategory = (category) => {
    // Formats category heading to camel case
    const formattedCategory = category
      .toLowerCase()
      .replace(/ ./g, (str) => str[1].toUpperCase());

    setDisplayCategory(formattedCategory);
    isChangingCategory.current = true;

    // Called after changing category animation has completed
    setTimeout(() => {
      isChangingCategory.current = false;
    }, 1000);
  };

  useEffect(() => {
    if (isMatch1120) {
      setProductsPerRow(4);
    } else if (isMatch845) {
      setProductsPerRow(3);
    } else {
      setProductsPerRow(2);
    }
  }, [isMatch845, isMatch1120]);

  return (
    <section>
      <Categories onChangeCategory={handleChangeCategory} />
      {offers[displayCategory]?.map((row, index) => (
        <Styled.Container key={index} data-testid={displayCategory}>
          <AnimatePresence
            initial={false}
            custom={isChangingCategory.current}
            mode={isChangingCategory.current ? 'wait' : 'sync'}
          >
            {row.map((product, productIndex) => {
              if (productIndex < productsPerRow) {
                return (
                  <MotionProductCard
                    key={product.sku}
                    custom={isChangingCategory.current}
                    variants={variants}
                    initial="enter"
                    animate="visible"
                    exit="exit"
                    product={product}
                  />
                );
              }
              return undefined;
            })}
          </AnimatePresence>
        </Styled.Container>
      ))}
    </section>
  );
}
