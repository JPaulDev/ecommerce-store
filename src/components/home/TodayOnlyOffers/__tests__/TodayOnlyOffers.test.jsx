import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithProviders } from '../../../../../test-utils';
import TodayOnlyOffers from '../index';

describe('TodayOnlyOffers component changing category', () => {
  beforeEach(() => {
    renderWithProviders(<TodayOnlyOffers />);
  });

  it('changes to graphics cards and renders 2 rows of products', () => {
    const button = screen.getByRole('button', {
      name: /graphics cards/i,
    });

    expect(screen.queryByTestId('graphicsCards')).not.toBeInTheDocument();

    userEvent.click(button);

    expect(screen.getAllByTestId('graphicsCards')).toHaveLength(2);
  });

  it('changes to memory and renders 2 rows of products', () => {
    const button = screen.getByRole('button', {
      name: /memory/i,
    });

    expect(screen.queryByTestId('memory')).not.toBeInTheDocument();

    userEvent.click(button);

    expect(screen.getAllByTestId('memory')).toHaveLength(2);
  });

  it('changes to hard drives and renders 2 rows of products', () => {
    const button = screen.getByRole('button', {
      name: /storage/i,
    });

    expect(screen.queryByTestId('storage')).not.toBeInTheDocument();

    userEvent.click(button);

    expect(screen.getAllByTestId('storage')).toHaveLength(2);
  });

  it('changes to motherboards and renders 2 rows of products', () => {
    const button = screen.getByRole('button', {
      name: /motherboards/i,
    });

    expect(screen.queryByTestId('motherboards')).not.toBeInTheDocument();

    userEvent.click(button);

    expect(screen.getAllByTestId('motherboards')).toHaveLength(2);
  });

  it('changes to processors and renders 2 rows of products', () => {
    const graphicsCards = screen.getByRole('button', {
      name: /graphics cards/i,
    });
    const processors = screen.getByRole('button', {
      name: /processors/i,
    });

    userEvent.click(graphicsCards);

    expect(screen.queryByTestId('processors')).not.toBeInTheDocument();

    userEvent.click(processors);

    expect(screen.getAllByTestId('processors')).toHaveLength(2);
  });
});

describe('TodayOnlyOffers component at different screen widths', () => {
  it('renders 2 rows with 4 products in each at 1120px and above', () => {
    window.resizeTo(1120, 768);
    renderWithProviders(<TodayOnlyOffers />);

    expect(screen.getAllByTestId('product')).toHaveLength(8);
  });

  it('renders 2 rows with 3 products in each at 1119px and below', () => {
    window.resizeTo(1119, 768);
    renderWithProviders(<TodayOnlyOffers />);

    expect(screen.getAllByTestId('product')).toHaveLength(6);
  });

  it('renders 2 rows with 2 products in each at 844px and below', () => {
    window.resizeTo(844, 768);
    renderWithProviders(<TodayOnlyOffers />);

    expect(screen.getAllByTestId('product')).toHaveLength(4);
  });
});
