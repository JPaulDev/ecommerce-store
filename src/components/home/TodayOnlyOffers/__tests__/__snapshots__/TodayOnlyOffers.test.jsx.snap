// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayOnlyOffers component should render with a category selector 1`] = `
<section>
  <div
    className="sc-2872bedc-0 daJcJp"
  >
    <ul
      className="sc-2872bedc-1 karVK"
    >
      <li>
        <button
          className="sc-2872bedc-2 dTNLyi"
          onClick={[Function]}
          type="button"
        >
          <div
            className="sc-2872bedc-3 hpYsdA"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 170 109.1"
              width="60%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M141.2,49.5c5.2.7,10-1.3,12.5-2.7a23.7,23.7,0,0,0-5.6-13,50.8,50.8,0,0,1-11.6,6.6c3.7-4,4.7-9.2,4.9-12a23.8,23.8,0,0,0-17.7-2.1,50.6,50.6,0,0,1,.7,13.4c-1.8-5-6-8.2-8.4-9.8a23.8,23.8,0,0,0-9.2,13.6,50.2,50.2,0,0,1,12.3,5.8c-5.5-.8-10.4,1.5-12.8,2.9a23.7,23.7,0,0,0,5.9,12.8,50.6,50.6,0,0,1,11.7-6.7c-3.6,3.9-4.6,9.1-4.9,11.9a23.8,23.8,0,0,0,16.5,2.1,50.8,50.8,0,0,1-.6-13.3c1.9,5.1,6.1,8.4,8.5,9.9A23.8,23.8,0,0,0,153,55.2,51.1,51.1,0,0,1,141.2,49.5Z"
              />
              <path
                d="M18,10.4V0H0V4.6H13.3V20.9H7V47h6.4v62.1H18V88.2H42.9v7H54.5v-7h5.8v7h77.7v-7H170V10.4ZM130,78.9a29.6,29.6,0,1,1,29.6-29.6A29.6,29.6,0,0,1,130,78.9Z"
              />
            </svg>
          </div>
          <h3
            className="sc-2872bedc-4 CVSHM"
          >
            Graphics Cards
          </h3>
        </button>
      </li>
      <li>
        <button
          className="sc-2872bedc-2 dTNLyi"
          onClick={[Function]}
          type="button"
        >
          <div
            className="sc-2872bedc-3 hpYsdA"
          >
            <svg
              fill="currentColor"
              viewBox="-18 0 606 512"
              width="60%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 448h80v-32c0-8.838 7.164-16 16-16c8.838 0 16 7.162 16 16v32h96v-32c0-8.838 7.164-16 16-16c8.838 0 16 7.162 16 16v32h96v-32c0-8.838 7.164-16 16-16c8.838 0 16 7.162 16 16v32h96v-32c0-8.838 7.164-16 16-16c8.838 0 16 7.162 16 16v32H576v-96H0V448zM576 146.9V112C576 85.49 554.5 64 528 64h-480C21.49 64 0 85.49 0 112v34.94C18.6 153.5 32 171.1 32 192S18.6 230.5 0 237.1V320h576V237.1C557.4 230.5 544 212.9 544 192S557.4 153.5 576 146.9zM192 240C192 248.8 184.8 256 176 256h-32C135.2 256 128 248.8 128 240v-96C128 135.2 135.2 128 144 128h32C184.8 128 192 135.2 192 144V240zM320 240C320 248.8 312.8 256 304 256h-32C263.2 256 256 248.8 256 240v-96C256 135.2 263.2 128 272 128h32C312.8 128 320 135.2 320 144V240zM448 240C448 248.8 440.8 256 432 256h-32C391.2 256 384 248.8 384 240v-96C384 135.2 391.2 128 400 128h32C440.8 128 448 135.2 448 144V240z"
              />
            </svg>
          </div>
          <h3
            className="sc-2872bedc-4 CVSHM"
          >
            Memory
          </h3>
        </button>
      </li>
      <li>
        <button
          className="sc-2872bedc-2 dTNLyi"
          onClick={[Function]}
          type="button"
        >
          <div
            className="sc-2872bedc-3 hpYsdA"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 169 112.7"
              width="60%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128.4,80.5a9.6,9.6,0,0,0,5.5-16.9A175.2,175.2,0,0,0,90.4,39c15,26.8,30.7,39,30.9,39.1h.1A9.6,9.6,0,0,0,128.4,80.5Z"
              />
              <circle
                cx="68.7"
                cy="56.3"
                r="9"
              />
              <path
                d="M133.9,63.6A156.5,156.5,0,0,0,113.3,50a44.9,44.9,0,0,1-1.4,19.2,84.8,84.8,0,0,0,9.3,9h.1a9.6,9.6,0,1,0,12.5-14.6Z"
              />
              <path
                d="M0,0V112.7H169V0ZM139.2,80.8a15.1,15.1,0,0,1-10.4,5.3h-1.2a15.1,15.1,0,0,1-9.8-3.6,81.5,81.5,0,0,1-8.1-7.5,45.1,45.1,0,1,1,1.9-32.5,163.1,163.1,0,0,1,25.8,16.7,15.2,15.2,0,0,1,1.7,21.5Z"
              />
            </svg>
          </div>
          <h3
            className="sc-2872bedc-4 CVSHM"
          >
            Storage
          </h3>
        </button>
      </li>
      <li>
        <button
          className="sc-2872bedc-2 dTNLyi"
          onClick={[Function]}
          type="button"
        >
          <div
            className="sc-2872bedc-3 hpYsdA"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 170 170"
              width="60%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0,0V170H170V0ZM149.9,138.7a11.2,11.2,0,0,1-11.2,11.2H31.3a11.2,11.2,0,0,1-11.2-11.2V31.3A11.2,11.2,0,0,1,31.3,20.1H138.7a11.2,11.2,0,0,1,11.2,11.2Z"
              />
              <polygon
                points="75.6 44.2 102.7 44.2 85 79.6 105.1 74.9 72 130.3 80.3 94.9 65 98.5 75.6 44.2"
              />
            </svg>
          </div>
          <h3
            className="sc-2872bedc-4 CVSHM"
          >
            Processors
          </h3>
        </button>
      </li>
      <li>
        <button
          className="sc-2872bedc-2 dTNLyi"
          onClick={[Function]}
          type="button"
        >
          <div
            className="sc-2872bedc-3 hpYsdA"
          >
            <svg
              fill="currentColor"
              viewBox="0 0 50 50"
              width="60%"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 18 4 L 18 6 L 7 6 C 5.346 6 4 7.346 4 9 L 4 43 C 4 44.654 5.346 46 7 46 L 43 46 C 44.654 46 46 44.654 46 43 L 46 9 C 46 7.346 44.654 6 43 6 L 39 6 L 39 4 L 26 4 L 26 6 L 24 6 L 24 4 L 18 4 z M 20 6 L 22 6 L 22 11 L 20 11 L 20 6 z M 28 6 L 37 6 L 37 11 L 28 11 L 28 6 z M 8 9 C 8.552 9 9 9.448 9 10 C 9 10.552 8.552 11 8 11 C 7.448 11 7 10.552 7 10 C 7 9.448 7.448 9 8 9 z M 42 9 C 42.552 9 43 9.448 43 10 C 43 10.552 42.552 11 42 11 C 41.448 11 41 10.552 41 10 C 41 9.448 41.448 9 42 9 z M 12 17 L 15 17 L 15 28 L 12 28 L 12 17 z M 17 17 L 20 17 L 20 28 L 17 28 L 17 17 z M 28 17 L 39 17 L 39 28 L 28 28 L 28 17 z M 31 20 L 31 25 L 36 25 L 36 20 L 31 20 z M 11 33 L 31 33 L 31 35 L 11 35 L 11 33 z M 37 34 C 38.103 34 39 34.897 39 36 C 39 37.103 38.103 38 37 38 C 35.897 38 35 37.103 35 36 C 35 34.897 35.897 34 37 34 z M 11 37 L 31 37 L 31 39 L 11 39 L 11 37 z M 8 41 C 8.552 41 9 41.448 9 42 C 9 42.552 8.552 43 8 43 C 7.448 43 7 42.552 7 42 C 7 41.448 7.448 41 8 41 z M 42 41 C 42.552 41 43 41.448 43 42 C 43 42.552 42.552 43 42 43 C 41.448 43 41 42.552 41 42 C 41 41.448 41.448 41 42 41 z"
              />
            </svg>
          </div>
          <h3
            className="sc-2872bedc-4 CVSHM"
          >
            Motherboards
          </h3>
        </button>
      </li>
    </ul>
  </div>
  <ul
    className="sc-2872bedc-5 cBWklk"
  />
</section>
`;
